### ISSUE TITLE
ATLauncher breaks on NAT64 networks

### ISSUE THREAD

**Issue by asdfjkluiop** [ATLauncher/ATLauncher#434]
**Describe the bug**
ATLauncher does not work properly on NAT64 networks because it forces java.net.preferIPv4Stack to true.

**To Reproduce**
Run ATLauncher on a pure IPv6 NAT64 network.

**Expected behavior**
ATLauncher should work properly under NAT64 if -Djava.net.preferIPv6Addresses=true is set from the command line.

**Additional context**
Ideally java.net.preferIPv4Stack should not be set as forcing IPv4 will become a bigger problem as the world transitions to IPv6, OpenJDK prefers v4 by default right now anyway, and manually forcing it breaks the ability to set java.net.preferIPv6Addresses=true and as a result breaks the launcher on NAT64 networks.

**Comment by RyanTheAllmighty** [comment]
That was added back in 2013 for some reason, I believe without it, it caused crashing in Minecraft at the time, would need to check and make sure why it was added

**Comment by jamierocks** [comment]
https://github.com/Slowpoke101/FTBLaunch/issues/637 was the relevant FTB issue, I assume it was the same case here.

**Comment by RyanTheAllmighty** [comment]
Yeah. I'd need to check everything we use is ipv6 compliant. I believe all the launcher and cdn services are, but I can't speak for everything else. There's also all the other services we use, CurseForge, Modpacks.ch, Fabric and Forge. So even if it's worked around to allow IPV6, it may not actually work.

Solution likely is unless `preferIPv6Addresses` is set, we'll preferIPV4

**Comment by asdfjkluiop** [comment]
While a lot of other services might not support v6 they don't have to if the user is on a NAT64 network for v6 to be used. NAT64 makes everything accessible over v6 even if it normally wouldn't be. The solution of checking for preferIPv6Addresses sounds like a perfectly good fix.

**Comment by RyanTheAllmighty** [comment]
Released with 3.4.2.6
